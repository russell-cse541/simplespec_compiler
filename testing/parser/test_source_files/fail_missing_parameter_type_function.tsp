//Parameter b in the section function is missing a type.

namespace  Main;

using Locale;
using Big;

scalar ID = ASCIIString {
    requires: value.length();
}

scalar Apple = ASCIIString {
    requires: component.juice();
    requires: component.seed();
}

model User {
    id: ID;
    name: String;
    region: Locale::Country;
    team: List<User,Hey>;
    fan: Bad<Yes>;
    requires: true;
    requires: a.torch();    
}

scalar Apple = ID;

model Test {}

const a:Bool = true;
const sillString:String = 'Yes This is a very Simple String!';

function checkit(): Bool {
    let b:Bool = true;
    let a:String = a.length();
    return u.a.length(pred(a:String): Bool => true);
}

function checkit(a:String, b, c:SomethingFake): Bool {
    return !(5>8) || (!(a || b && c));
}