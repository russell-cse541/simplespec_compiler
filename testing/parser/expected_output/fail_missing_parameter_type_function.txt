Error: syntax error line: 40
Error: [1;31m Improper Declaration [0m line: 41
Error: syntax error line: 42


[0;33m There were errors parsing the document. Start by fixing the initial error.  Other errors may be uncovered as some are corrected.[0m

